packages/bitcore-node/src/models/transaction.ts

```diff
@@ -254,7 +254,7 @@ export class TransactionModel extends BaseModel<ITransaction> {
         if (!agg[mintTxid]) {
           agg[mintTxid] = {
             total: value,
-            wallets: wallets || []
+            wallets: [...wallets] || []
```

Comment:
Before this change, this was by reference, so the following pushes would mutate the original wallets array.